pkg/operator/configobserver/cloudprovider/observe_cloudprovider.go

```diff
@@ -72,13 +75,25 @@ func (c *cloudProviderObserver) ObserveCloudProviderNames(genericListers configo
 
 	sourceCloudConfigMap := infrastructure.Spec.CloudConfig.Name
 	sourceCloudConfigNamespace := configNamespace
+	sourceCloudConfigKey := infrastructure.Spec.CloudConfig.Key
+
+	// If a managed cloud-provider config is available, it should be used instead of the default. If the configmap is not
+	// found, the default values should be used.
+	if _, err = listers.ConfigMapLister().ConfigMaps(machineSpecifiedConfigNamespace).Get(machineSpecifiedConfig); err == nil {
+		sourceCloudConfigMap = machineSpecifiedConfig
+		sourceCloudConfigNamespace = machineSpecifiedConfigNamespace
+		sourceCloudConfigKey = "cloud.conf"
+	} else if !errors.IsNotFound(err) {
+		return existingConfig, append(errs, err)
+	}
+
 	sourceLocation := resourcesynccontroller.ResourceLocation{
 		Namespace: sourceCloudConfigNamespace,
 		Name:      sourceCloudConfigMap,
 	}
 
 	// we set cloudprovider configmap values only for some cloud providers.
-	validCloudProviders := sets.NewString("azure", "gce", "openstack", "vsphere")
+	validCloudProviders := sets.NewString("aws", "azure", "gce", "openstack", "vsphere")
```

Comment:
Hm I need to make sure this is ok. I added "aws" because in the case of custom region endpoints, aws will have a cloud config, but otherwise won't.

Perhaps that missing cloud config will pose a problem now that aws is in the list. I am running another e2e-aws test in openshift/cluster-kube-controller-manager-operator#430 but perhaps someone may know off the top of their head

Reply:
I tested locally on aws and everything looked fine. Also e2e-aws on dependent pr openshift/cluster-kube-controller-manager-operator#430 passed so everything looks good to me.